model/monitorCache/implementation/DeliveryMonitoringData.php

```diff
@@ -263,10 +270,129 @@ private function getEndTime()
+     * @return string
+     */
+    private function getTestTakerLastName(){
+        return UserHelper::getUserLastName($this->getUser());
+    }
+
+    /**
+     * @param bool $overwrite
+     */
+    private function addExtraFieldsValues($overwrite = false)
+    {
+        $user = $this->getUser();
+        if ($user) {
+            $fields = DeliveryHelper::getExtraFieldsProperties();
+            foreach ($fields as $field) {
+
+                $values = $user->getPropertyValues($field['property']);
+                if (!empty($values) && is_array($values)) {
+                    $this->addValue($field['id'], (string)$values[0], $overwrite);
+                }
+            }
+        }
+    }
+
+    /**
+     * @return string
+     */
+    private function getTestCenterUri()
+    {
+        $uri = null;
+        $delivery = $this->deliveryExecution->getDelivery();
+        $user = $this->getUser();
+
+        $testCenter = EligibilityService::singleton()->getTestCenter($delivery, $user);
+        if ($testCenter) {
+            $uri = $testCenter->getUri();
+        } else {
+            $deliverLog = ServiceManager::getServiceManager()->get(DeliveryLog::SERVICE_ID);
+            $loggedEvent = $deliverLog->get(
+                $this->deliveryExecution->getIdentifier(),
+                'TEST_AUTHORISE'
+            );
+            $loggedEvent = reset($loggedEvent);
+            $uri = isset($loggedEvent['data']['test_center']) ? $loggedEvent['data']['test_center'] : null;
+        }
+        return $uri;
+    }
+
+    /**
+     * Set test session
+     * @param AssessmentTestSession $testSession
+     */
+    public function setTestSession(AssessmentTestSession $testSession)
+    {
+        $this->testSession = $testSession;
+        $this->updateData();
+    }
+
+    private function getTestSession()
+     {
+         if ($this->testSession === null) {
+             $testSessionService = TestSessionService::singleton();
+             $this->testSession = $testSessionService->getTestSession($this->deliveryExecution);
+         }
+        return $this->testSession;
+     }
+    
+
+    /**
+     * Refresh delivery information in storage
+     */
+    public function updateDeliveryLabel()
+    {
+        $this->addValue(DeliveryMonitoringService::DELIVERY_NAME, $this->deliveryExecution->getDelivery()->getLabel(), true);
+    }
+
+    /**
+     * Refresh testtaker information in storage
+     */
+    public function updateTestTakerData()
+    {
+        $this->addValue(DeliveryMonitoringService::TEST_TAKER, $this->getTestTaker(), true);
+        $this->addValue(DeliveryMonitoringService::TEST_TAKER_FIRST_NAME, $this->getTestTakerFistName(), true);
+        $this->addValue(DeliveryMonitoringService::TEST_TAKER_LAST_NAME, $this->getTestTakerLastName(), true);
+        $this->addExtraFieldsValues(true);
+    }
+
+    /**
+     * @param bool|timestamp $lastConnectivity
+     */
+    public function updateStatus($lastConnectivity = false)
```

Comment:
I reviewed and tested it and pull request looks good for me. Only one note:
Please add three last public method to the `DeliveryMonitoringData` interface.